activerecord/test/cases/relations_test.rb

```diff
@@ -370,6 +370,20 @@ def test_default_reverse_order_on_table_without_primary_key
     end
   end
 
+  def test_default_reverse_order_on_table_without_primary_key_with_implicit_order_column
+    implicit_order_column_before = Edge.implicit_order_column
+    Edge.implicit_order_column = "source_id"
+
+    Edge.delete_all
+
+    edge_1 = Edge.create!(source_id: 1, sink_id: 2)
+    edge_2 = Edge.create!(source_id: 2, sink_id: 3)
+    assert_equal edge_1.source_id, Edge.all.first.source_id
+    assert_equal edge_2.source_id, Edge.all.reverse_order.first.source_id
+  ensure
+    Edge.implicit_order_column = implicit_order_column_before
+  end
```

Comment:
Rather than to mutate the model, you can do an anonymous subclass of it:

```suggestion
  def test_default_reverse_order_on_table_without_primary_key_with_implicit_order_column
    ordered_edge = Class.new(Edge) do
      self.implicit_order_column = "source_id"
    end

    ordered_edge.delete_all

    edge_1 = ordered_edge.create!(source_id: 1, sink_id: 2)
    edge_2 = ordered_edge.create!(source_id: 2, sink_id: 3)
    assert_equal edge_1.source_id, ordered_edge.all.first.source_id
    assert_equal edge_2.source_id, ordered_edge.all.reverse_order.first.source_id
  end
```

Reply:
Addressed 👍🏽